### PR TITLE
Fix some SciPy broken links

### PR DESCRIPTION
**Pull request by astrojuanlu**
While I was thinking about addressing #912, I decided to start with a smaller pull request first to get familiar with the contribution workflow and the expectations from reviewers.

This amends several build warnings I found because the SciPy website has changed a lot quite recently.

### CODE REVIEW

File: source/conf.py [hunk 1]
```diff
@@ -419,7 +419,6 @@
     'python': ('https://docs.python.org/3', None),
     'python-guide': ('https://docs.python-guide.org', None),
     'python2': ('https://docs.python.org/2', None),
-    'scipy': ('https://www.scipy.org', None),
```
Comment (webknjaz) [inline source/conf.py:422]: Looks like this should be https://scipy.github.io/devdocs/ now. I've updated it here https://webknjaz.github.io/intersphinx-untangled/scipy.github.io/ a while ago.

File: source/conf.py [hunk 2]
```diff
@@ -419,7 +419,6 @@
...
     'setuptools': ('https://setuptools.readthedocs.io/en/latest/', None),
```
Comment (webknjaz) [inline source/conf.py:422]:
```suggestion
    'scipy': ('https://scipy.github.io/devdocs', None),
    'setuptools': ('https://setuptools.readthedocs.io/en/latest/', None),
```

File: source/contribute.rst [hunk 3]
```diff
@@ -142,7 +142,7 @@ To build the guide, run the following bash command in the source folder:
    nox -s build
 
 After the process has completed you can find the HTML output in the
-``./build/html`` directory. You can open the ``index.html`` file to view the
```
Comment (webknjaz) [inline source/contribute.rst:145]: This is unrelated, submit separately.

**Review by webknjaz — CHANGES_REQUESTED** [review]
Please keep intersphinx references where possible.

**Comment by astrojuanlu** [comment]
> Please keep intersphinx references where possible.

I don't think that's possible, but I'll leave individual comments on each of the references I changed to discuss.

File: source/conf.py (hunk 1 above)
Comment (astrojuanlu) [reply source/conf.py:422]: (Postponing replying to this until we decide whether we need it or not)

File: source/guides/installing-scientific-packages.rst [hunk 4]
```diff
@@ -14,8 +14,8 @@ kinds of hardware, or to interoperate with different pieces of external
 software.
 
 In particular, `NumPy <https://numpy.org/>`__, which provides the basis
-for most of the software in the :doc:`scientific Python stack
-<scipy:stackspec>` can be configured
+for most of the software in the `scientific Python stack
+<https://scientific-python.org>`_ can be configured
```
Comment (astrojuanlu) [inline source/guides/installing-scientific-packages.rst:18]: The `stackspec` page is gone. Last time seen: https://web.archive.org/web/20211023180926/https://www.scipy.org/stackspec.html

With a note that says:

> The SciPy Stack specification was developed in 2012. As of 2017, the SciPy Stack concept is obsolete given improvements in package management and distribution.

I replaced it with a reference to `scientific-python.org`, which is not a Sphinx site (I believe it's based on Hugo?) and so I can't use an intersphinx reference here.

File: source/guides/installing-scientific-packages.rst [hunk 5]
```diff
@@ -90,16 +90,16 @@ macOS installers that are compatible with the macOS CPython binaries
...
+`install SciPy on macOS <https://scipy.org/install/#mac-packages>`_.
 
 
 SciPy distributions
 -------------------
 
-The SciPy site lists :doc:`several distributions
-<scipy:install>` that provide the full SciPy stack to
+The SciPy site lists `several distributions
+<https://scipy.org/install/#distributions>`_
```
Comment (astrojuanlu) [inline source/guides/installing-scientific-packages.rst:101]: Same page as above, same reasoning.

File: source/guides/installing-scientific-packages.rst (hunk 4 above)
Comment (webknjaz) [reply source/guides/installing-scientific-packages.rst:18]: Okay, that makes sense. (for non-sphinx sites, it's possible to create roles but this would be an overkill in our case)

File: source/guides/installing-scientific-packages.rst [hunk 6]
```diff
@@ -90,16 +90,16 @@ macOS installers that are compatible with the macOS CPython binaries
 published on python.org.
 
 macOS users also have access to Linux distribution style package managers
-such as ``MacPorts``. The SciPy site has more details on using MacPorts to
-install the :ref:`scientific Python stack
-<scipy:mac>`
+such as ``Homebrew``. The SciPy site has more details on using Homebrew to
+`install SciPy on macOS <https://scipy.org/install/#mac-packages>`_.
```
[the remark this answers is not in the document]
Comment (webknjaz) [reply source/guides/installing-scientific-packages.rst:94]: Okay, looks like the Sphinx docs don't have equivalent info.

File: source/guides/installing-scientific-packages.rst (hunk 5 above)
Comment (webknjaz) [reply source/guides/installing-scientific-packages.rst:101]: Alright, makes sense.

**Review by webknjaz — APPROVED** [review]
All's good. Thanks!

File: source/guides/installing-scientific-packages.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (webknjaz) [reply source/guides/installing-scientific-packages.rst:94]: Linkcheck says there's no `#mac-packages` anchor on that page.

[the remark this answers is not in the document]
Comment (webknjaz) [reply source/guides/installing-scientific-packages.rst:94]: Aha! It's just `#mac` in HTML. `#mac-packages` is a dynamic thing that only appears in JS runtime and the linter can't catch that.
```suggestion
`install SciPy on macOS <https://scipy.org/install/#mac>`_.
```

**Comment by astrojuanlu** [comment]
Thanks for the quick turnaround @webknjaz !
